### PR TITLE
Add sccache support

### PR DESCRIPTION
**Pull request by zachhannum**
### Intent

Addresses #12520. Adds support for sccache to the build. Set the environment variable SCCACHE_ENABLED=1 to use sccache, otherwise sccache will not be used. The pull-request Jenkinsfile has also been updated to use sccache.

### Approach

An [S3 bucket has been created](https://s3.console.aws.amazon.com/s3/buckets/rstudio-build-cache?region=us-east-1&tab=objects) to act as a build cache for Jenkins (and developers, if they so choose). Note that this does not add sccache for Windows or OSX at this time. We should create new issues to address this if needed.

### Automated Tests

N/A

### QA Notes

N/A

### Documentation
N/A

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

File: cmake/modules/FindSccache.cmake [hunk 1]
```diff
@@ -0,0 +1,26 @@
+find_program(CCACHE_PROGRAM sccache)
```
Comment (melissa-barca) [inline cmake/modules/FindSccache.cmake:1]: This file looks very similar to sccache.cmake. Why are both necessary?

File: dependencies/common/install-sccache [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+#
+# install-sccache
+#
+# Copyright (C) 2022 by RStudio, PBC
```
Comment (melissa-barca) [inline dependencies/common/install-sccache:6]: We 
```suggestion
# Copyright (C) 2023 by RStudio, PBC
```

We'll need to go through and update our copyrights with the new year, but best not to introduce any new 2022 Copyrights.

File: cmake/modules/FindSccache.cmake (hunk 1 above)
Comment (zachhannum) [reply cmake/modules/FindSccache.cmake]: Oops, good catch. You're right, this was a copy/paste error. Removed.

File: dependencies/common/install-sccache (hunk 2 above)
Comment (zachhannum) [reply dependencies/common/install-sccache]: Updated!

File: dependencies/common/install-sccache [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+#
+# install-sccache
+#
+# Copyright (C) 2023 by RStudio, PBC
```
Comment (melissa-barca) [inline dependencies/common/install-sccache:6]:
```suggestion
# Copyright (C) 2023 by Posit Software, PBC
```

Sorry, I should've caught this the first time! This also needs to be Posit Software. I just `grep`-ed the repo and it looks like the majority of copyrights were updated but a few in `dependencies` were missed.

**Review by melissa-barca — APPROVED** [review]
LGTM!

File: dependencies/common/install-sccache (hunk 3 above)
Comment (zachhannum) [reply dependencies/common/install-sccache]: Went ahead and updated the copyright notices in `dependencies` to use Posit!

Comment (melissa-barca) [reply dependencies/common/install-sccache]: Nice, thank you!

File: package/linux/make-package [hunk 4]
```diff
@@ -225,6 +225,20 @@ if [ -n "${JENKINS_URL}" ]; then
    printenv
 fi
 
+# if SCCACHE_ENABLED environment variable is set, use sccache
+if [ -n "$SCCACHE_ENABLED" ]; then
+    echo "Using sccache"
+   
+    if [ -n "$AWS_ACCESS_KEY_ID" ] || [ $(aws sts get-caller-identity --query "Account" --profile ${AWS_PROFILE:-sso}) -eq 14 ]; then
```
Comment (cm421) [inline package/linux/make-package:232]: Do any of these need to be pre-defined before this script runs? If so we'll want to add that info into the wiki

**Review by cm421 — CHANGES_REQUESTED** [review]
Sorry for not getting these in before we merged, but I think the dependency script needs some attention

File: dependencies/common/install-sccache [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+ARCH=x86_64
+SCCACHE_PACKAGE_HASH="94ea33aac8dcb358753f8240cc87345963cf83cda7c6af0395dff31ffdc88df4"
+
+# override to aarch64 if running on arm
+if [ "$(arch)" = "aarch64" ] ; then
+    ARCH=aarch64
+    SCCACHE_PACKAGE_HASH="1bf58385dc27b66324bb9ee82084e65c4d2e60baa19e3d16d2ab4da6c1ae66b2"
+fi
+
+URL=https://github.com/mozilla/sccache/releases/download/v${VERSION}/sccache-v${VERSION}-${ARCH}-unknown-linux-musl.tar.gz
```
[the remark this answers is not in the document]
Comment (zachhannum) [reply dependencies/common/install-sccache:41]: I created #12533 and #12532 to address the sccache dependency for macOS and Linux, but I will create a small patch to move this script for now!

**Pull request merged**